tf2_ros/include/tf2_ros/static_transform_component.hpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef TF2_ROS__STATIC_TRANSFORM_COMPONENT_HPP_
+#define TF2_ROS__STATIC_TRANSFORM_COMPONENT_HPP_
+
+#include "rclcpp/clock.hpp"
+#include "rclcpp/time_source.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include <tf2/LinearMath/Quaternion.h>
+#include "tf2_ros/static_transform_broadcaster.h"
+
+#include "builtin_interfaces/msg/time.hpp"
+
+
+namespace tf2_ros
+{
+class static_transform_component : public rclcpp::Node
+{
+public:
+  explicit static_transform_component(
+    double tx, double ty, double tz,
+    double ax, double ay, double az, double aw,
+    const char * frame_id, const char * child_id,
+    rclcpp::NodeOptions options, const char * name);
+
+  explicit static_transform_component(
+    double tx, double ty, double tz,
+    double r, double p, double y,
+    const char * frame_id, const char * child_id,
+    rclcpp::NodeOptions options, const char * name);
```

Comment:
I don't think that these prototypes can actually be called in the component case.  How would we initialize them in that case?  We may have to think about using parameters instead here.

Reply:
>  How would we initialize them in that case? 

That's a valid point. I added the `send_transform` function as an attempt to cover this case, but I'm not convinced that's valid.

Those cases are somewhat useful for use with an executor, so I'd like to avoid removing them entirely, but I am not sure what the right thing to do here is.

I'll do the other cleanup items while we figure out the best way to deal with this.